ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -579,19 +590,25 @@ applyPatchCmmParseNoImplicitPrelude ghcFlavor =
 -- function 'appyPatchHadrianStackYaml' guarantees it is available
 -- if its needed.
 
--- Patch Hadrian's Cabal.
+-- | Patch Hadrian's Cabal.
 applyPatchHadrianStackYaml :: GhcFlavor -> IO ()
 applyPatchHadrianStackYaml ghcFlavor = do
-   appendFile "hadrian/stack.yaml" $ unlines ["ghc-options:","  \"$everything\": -O0 -j"]
-   case ghcFlavor of
-     GhcMaster ->
-       appendFile "hadrian/stack.yaml" $
-       unlines [
-           "extra-deps:"
-         , "  - happy-1.20.0" -- See https://gitlab.haskell.org/ghc/ghc/-/issues/18726.
-         , "  - exceptions-0.10.4" -- See [Note : GHC now depends on exceptions package]
-         ]
-     _ -> pure ()
+  let hadrianStackYaml = "hadrian/stack.yaml"
+  bytes <- B.readFile hadrianStackYaml
+  config <- Y.decodeThrow bytes
+  -- See [Note : GHC now depends on exceptions package]
+  let deps = [x | ghcFlavor == GhcMaster, x <- [T.pack "exceptions-0.10.4"]] ++
+        case parse (\cfg -> cfg .:? "extra-deps" .!= []) config of
+          Success ls -> ls :: [T.Text]
+          _ -> []
```

Comment:
If `extra-deps` is not a list, something must have gone wrong. Let's not swallow the error but rather crash noisily.
```suggestion
          Error msg -> error msg
```

Reply:
Ok